scripts/prepare_perfect_foresight.py

```diff
@@ -487,6 +488,38 @@ def apply_time_segmentation_perfect(
     return n
 
 
+def update_heat_pump_efficiency(n: pypsa.Network, years: List[int]):
```

Comment:
Couldn't this and the `update_heat_pump_efficiency` in rule `add_brownfield` be combined?

Reply:
Good point. Thought about making this a `_helper` function, too. But the structure is a little different between the two. 
I think it would boil down to making `n_p` an optional input and checking for `if n_p is None`, basically copying the current code into the `if` and `else` respectively. Happy to do that if you prefer it.